packages/block-library/src/block/index.php

```diff
@@ -22,6 +24,23 @@ function render_block_core_block( $attributes ) {
 		return '';
 	}
 
+	if ( in_array( $attributes['ref'], $seen_refs, true ) ) {
+		if ( ! is_admin() ) {
+			trigger_error(
+				sprintf(
+					// translators: %s is the user-provided title of the reusable block.
+					__( 'Could not render Reusable Block <strong>%s</strong>: blocks cannot be rendered inside themselves.' ),
+					$reusable_block->post_title
+				),
+				E_USER_WARNING
+			);
+		}
+
+		return '[block rendering halted]';
```

Comment:
Thoughts on leaving this here or returning `''`?

Reply:
I'd go for returning it, as it can be overlooked that there is a problem, by having the warnings not visible.

Reply:
I don't know if that is helpful for someone browsing the website. It's not that far to displaying an error when PHP can't connect with the database. You could display it maybe when `WP_DEBUG` is enabled but maybe with a ton of surrounding red colors 😄

Reply:
I went with `WP_DEBUG_DISPLAY`, per:

https://github.com/WordPress/wordpress-develop/blob/21b667b8405d8934c0ed8463e3273f3e4c22c989/src/wp-includes/load.php#L394-L400